### DISCUSSION TITLE
fission function create

### DISCUSSION THREAD

**Discussion by nxtcoder17** [fission/fission#2444] [Q&A] [answered]
**fission function create --buildcmd=**  does not take any arguments, i am kind of in a situation where i think the ability to pass ARGS, would have been better. 
 
I did take a look at the [pkg/builder source](https://github.com/fission/fission/blob/063209dc9ea535c6578580ba285c602fb5b55b72/pkg/builder/builder.go#L126), here it calls a function defined [here](https://github.com/fission/fission/blob/063209dc9ea535c6578580ba285c602fb5b55b72/pkg/builder/builder.go#L166). Inside this implementation, the team has used `exec(command)`, taking away the ability to pass any ARGS.

Is this done on purpose, and should i try some other way ?

**Comment by sanketsudake** [comment]
Hi @nxtcoder17 Did you try passing arguments in "" quotes along with command ?

**Reply by nxtcoder17** [reply]
yeah, i had tried this command `fission function create --name x57 --env notsample2 --src x.zip --buildcmd="/build sample1 sample2"`

but it throws an error in my builder

![err](https://user-images.githubusercontent.com/22402557/171173271-41a60325-28e0-4d31-8926-8a631b04e42a.png)

**Reply by nxtcoder17** [reply]
that is why i digged into the source code to find [this](https://github.com/fission/fission/blob/063209dc9ea535c6578580ba285c602fb5b55b72/pkg/builder/builder.go#L166), the function does not have any args parameter.

you can see on the [next line](https://github.com/fission/fission/blob/063209dc9ea535c6578580ba285c602fb5b55b72/pkg/builder/builder.go#L167), it instantiates command with `cmd := exec.Command(command)`, but to consider cli args, it should have been `cmd := exec.Command(command, args...)`

**Comment by sanketsudake** [comment, marked as answer]
This would get fixes as part of fix for issue https://github.com/fission/fission/issues/2445
